core-rust/mesh-api-server/src/mesh_api/conversions/block.rs

```diff
@@ -0,0 +1,71 @@
+use crate::prelude::*;
+
+pub fn extract_state_version_from_mesh_api_partial_block_identifier(
+    block_identifier: &models::PartialBlockIdentifier,
+) -> Result<Option<StateVersion>, ExtractionError> {
+    let state_version = match &block_identifier.hash {
```

Comment:
SUPER MINOR / optional: It might be a tiny bit easier to read / follow the different cases written like this (with some shared code for resolving hash and index)
```rust
match (block_identifier.index.as_ref(), block_identifier.hash.as_ref()) {
    (None, None) => {
        // ERROR
    }
    (Some(index), None) => {
        // Resolve index
    }
    (None, Some(hash)) => {
        // Resolve hash
    }
    (Some(index), Some(hash)) => {
        // Resolve index and hash; check they agree else error
    }
}
```

Reply:
Yeah, was trying this, but did not want to repeat code when parsing hash :)
But will rework it